zerver/lib/digest.py

```diff
@@ -166,36 +165,31 @@ def handle_digest_email(user_profile_id: int, cutoff: float,
                         render_to_web: bool = False) -> Union[None, Dict[str, Any]]:
     user_profile = get_user_profile_by_id(user_profile_id)
 
-    # We are disabling digest emails for soft deactivated users for the time.
-    # TODO: Find an elegant way to generate digest emails for these users.
-    if user_profile.long_term_idle:
-        return None
-
     # Convert from epoch seconds to a datetime object.
     cutoff_date = datetime.datetime.fromtimestamp(int(cutoff), tz=pytz.utc)
 
-    all_messages = UserMessage.objects.filter(
-        user_profile=user_profile,
-        message__pub_date__gt=cutoff_date
-    ).select_related('message').order_by("message__pub_date")
-
     context = common_context(user_profile)
 
     # Start building email template data.
     context.update({
         'unsubscribe_link': one_click_unsubscribe_link(user_profile, "digest")
     })
 
-    home_view_recipients = Subscription.objects.filter(
+    home_view_streams = Subscription.objects.filter(
         user_profile=user_profile,
         active=True,
```

Comment:
Let's add `recipient__type=Recipient.STREAM` to the filter here -- we don't want huddles etc. and `in_home_view=True` is sorta a weird way to express that.

Reply:
I guess the code here is super old, and hence does it that way. I'll change it!